src/g1.rs

```diff
@@ -751,6 +751,29 @@ impl G1Projective {
         G1Projective::conditional_select(&tmp, self, rhs.is_identity())
     }
 
+    /// Multiply this point by a scalar, using variable-time multiplication.
+    pub fn multiply_vartime(&self, by: &Scalar) -> G1Projective {
+        let mut acc = G1Projective::identity();
+
+        // This is a simple double-and-add implementation of point
+        // multiplication, moving from most significant to least
+        // significant bit of the scalar.
+        //
+        // We skip the leading bit as part of the vartime implementation.
+        for bit in by
+            .to_bytes()
+            .iter()
+            .rev()
+            .flat_map(|byte| (0..8).rev().map(move |i| Choice::from((byte >> i) & 1u8)))
+            .skip_while(|c| !bool::from(*c))
+        {
+            acc = acc.double();
+            acc = G1Projective::conditional_select(&acc, &(acc + self), bit);
```

Comment:
It seems like you could just use a branch here, rather than `conditional_select`, since this is variable-time code?

Likewise you could use a `bool` instead of `Choice`, which will avoid `subtle`'s memory barriers and should make the code easier for rustc to optimize.